internal/authz/oidc.go

```diff
@@ -396,6 +396,81 @@ func (o *oidcHandler) retrieveTokens(ctx context.Context, log telemetry.Logger,
 	setDenyResponse(resp, deny, codes.Unauthenticated)
 }
 
+// buildAuthHeader builds the authorization header for the client according to the
+// client authentication method specified in the OIDCConfig.
+//
+// The function returns an error if the client authentication method is unspecified
+// or if the implementation for the specified method is not supported.
+func buildAuthHeader(config *oidcv1.OIDCConfig) (http.Header, error) {
+
+	headers := http.Header{}
+	switch config.GetMethod() {
+	case oidcv1.OIDCConfig_CLIENT_AUTHENTICATION_METHOD_BASIC:
+	default:
+		headers = http.Header{
+			inthttp.HeaderContentType:   []string{inthttp.HeaderContentTypeFormURLEncoded},
+			inthttp.HeaderAuthorization: []string{inthttp.BasicAuthHeader(config.GetClientId(), config.GetClientSecret())},
+		}
+
+	case oidcv1.OIDCConfig_CLIENT_AUTHENTICATION_METHOD_CLIENT_SECRET_POST:
+
+		// Build post auth header
+		headers = http.Header{
+			inthttp.HeaderContentType: []string{inthttp.HeaderContentTypeFormURLEncoded},
+		}
+
+	case oidcv1.OIDCConfig_CLIENT_AUTHENTICATION_METHOD_CLIENT_SECRET_JWT:
+		// Build jwt auth header
+		// TODO: implement jwt auth header
+		return nil, errors.New("not implemented")
```

Comment:
Add more details in the error about _what_ is not implemented so users can understand what is the problem when receiving the response back.